sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/IcebergUtils.java

```diff
@@ -54,6 +58,14 @@ private IcebergUtils() {}
           .put(Schema.TypeName.DOUBLE, Types.DoubleType.get())
           .put(Schema.TypeName.STRING, Types.StringType.get())
           .put(Schema.TypeName.BYTES, Types.BinaryType.get())
+          .put(Schema.TypeName.DATETIME, Types.TimestampType.withoutZone())
+          .build();
+
+  private static final Map<String, Type> BEAM_LOGICAL_TYPES_TO_ICEBERG_TYPES =
+      ImmutableMap.<String, Type>builder()
+          .put(SqlTypes.DATE.getIdentifier(), Types.DateType.get())
+          .put(SqlTypes.TIME.getIdentifier(), Types.TimeType.get())
+          .put(SqlTypes.DATETIME.getIdentifier(), Types.TimestampType.withoutZone())
```

Comment:
Does that mean that beam doesn't support Timezone timestamps?

Reply:
Added support for timezone timestamps with strings

Reply:
Btw I noticed that when reading `timestamptz` types from Iceberg, the timestamp is always returned in UTC.

I wonder if it's something I'm doing on my side, or if this is expected Iceberg behavior

Reply:
Ah looks like this is standard Iceberg behavior. From [docs](https://iceberg.apache.org/spec/#primitive-types):

`Timestamp values with time zone represent a point in time: values are stored as UTC and do not retain a source time zone (2017-11-16 17:10:34 PST is stored/retrieved as 2017-11-17 01:10:34 UTC and these values are considered identical)`

Reply:
Yes, it's expected behaviour. 

Iceberg stores both TS and NTZ as UTC and it's up to the engine to do the required conversions from their internal TS representation. So, it is actually okay if it's NTZ.